From e775bc997a2a7605e5672fd0d78d5c266883af94 Mon Sep 17 00:00:00 2001
From: Artem Shinkarov <artyom.shinkaroff@gmail.com>
Date: Tue, 28 Apr 2015 14:27:59 +0100
Subject: [PATCH 2/2] Higher optimisation level when compiling i3.

---
 common.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common.mk b/common.mk
index dfa2731..918e134 100644
--- a/common.mk
+++ b/common.mk
@@ -1,5 +1,5 @@
 UNAME=$(shell uname)
-DEBUG=1
+DEBUG=0
 INSTALL=install
 LN=ln
 PKG_CONFIG=pkg-config
@@ -39,7 +39,7 @@ ifeq ($(DEBUG),1)
 # Extended debugging flags, macros shall be available in gcc
 CFLAGS ?= -pipe -gdwarf-2 -g3
 else
-CFLAGS ?= -pipe -O2 -freorder-blocks-and-partition
+CFLAGS ?= -pipe -O3 -march=native -mtune=native -freorder-blocks-and-partition
 endif
 
 # Default LDFLAGS that users should be able to override
-- 
2.3.6

