From 57097699cdd4f74d70838af6248cec31ca351e81 Mon Sep 17 00:00:00 2001
From: Artem Shinkarov <artyom.shinkaroff@gmail.com>
Date: Sat, 16 Nov 2013 17:02:30 +0000
Subject: [PATCH 2/2] Higher optimisation level when compiling i3.

---
 common.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common.mk b/common.mk
index 5581351..4744deb 100644
--- a/common.mk
+++ b/common.mk
@@ -1,5 +1,5 @@
 UNAME=$(shell uname)
-DEBUG=1
+DEBUG=0
 COVERAGE=0
 INSTALL=install
 LN=ln
@@ -36,7 +36,7 @@ ifeq ($(DEBUG),1)
 # Extended debugging flags, macros shall be available in gcc
 CFLAGS ?= -pipe -gdwarf-2 -g3
 else
-CFLAGS ?= -pipe -O2 -freorder-blocks-and-partition
+CFLAGS ?= -pipe -O3 -march=native -mtune=native -freorder-blocks-and-partition
 endif
 
 # Default LDFLAGS that users should be able to override
-- 
2.0.0

